keras/backend/tensorflow_backend.py

```diff
@@ -1866,6 +1866,26 @@ def batch_normalization(x, mean, var, beta, gamma, epsilon=1e-3):
     # Returns
         A tensor.
     """
+    if ndim(x) == 4:
+        # The CPU implementation of FusedBatchNorm only support NHWC
+        if len(mean.shape) == 1:
+            tf_data_format = 'NHWC'
```

Comment:
This is incorrect. What data format to use depends on what `axis` argument was passed to the BN layer.

Reply:
Agree. It's much more clean way to introduce the axis/reduction_axes parameter for backend interface function "batch_normalization" just as the "normalize_batch_in_training" function, then all the three backend cntk/theano/tensorflow will have axis info in batch_normalization function, though current theano/cntk don't need to deal axis info( I don't know whether they have fused BatchNorm version imp).

Previous I don't want to change the "batch_normalization" backend interface, and tried to get axis info from shape, the logic is little tricky there.